plugins/modules/activation.py

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+
+
+# (c) 2023, Nikhil Jain <nikjain@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = """
+---
+module: activation
+author: "Nikhil Jain (@jainnikhil30)"
+short_description: Create, restart or delete rulebook activations in EDA Controller.
+description:
+  - This module allows you to create, restart or delete activations in a EDA controller.
+options:
+  name:
+    description:
+      - The name of the rulebook activation.
+    type: str
+    required: true
+  description:
+    description:
+      - The description of the rulebook activation (optional).
+    type: str
+    required: false
+  project:
+    description:
+      - The name of the project associated with the rulebook activation.
+    type: str
+    required: true
+  rulebook:
+    description:
+      - The name of the rulebook associated with the rulebook activation.
+    type: str
+    required: true
+  extra_vars:
+    description:
+      - The extra variables for the rulebook activation, default is ''.
+    type: str
+    default: ''
+  restart_policy:
+    description:
+      - The restart policy for the rulebook activation, default is 'always'.
+    default: 'always'
+    choices: ["on-failure", "always", "never"]
+    type: str
+  enabled:
+    description:
+      - Whether the rulebook activation is enabled or not, default 'True'.
+    type: bool
+    default: true
+  decision_environment:
+    description:
+      - The name of the decision environment associated with the rulebook activation.
+    type: str
+    required: true
+  state:
+    description:
+      - Desired state of the resource.
+    default: "present"
+    choices: ["present", "absent", "exists"]
+    type: str
+extends_documentation_fragment: ansible.eda.auth
+"""
+
+EXAMPLES = """
+- name: Create EDA Activation
+  ansible.eda.activation:
+    name: "Example Activation"
+    description: "Example Activation description"
+    project: "Example Project"
+    rulebook: "hello_controller.yml"
+    decision_environment: "Default Decision Environment"
+    enabled: False
+
+- name: Delete EDA Activation
+  ansible.eda.activation:
+    name: "Example Activation"
+    description: "Example Activation description"
+    project: "Example Project"
+    rulebook: "hello_controller.yml"
+    decision_environment: "Default Decision Environment"
+    enabled: False
+    state: absent
```

Comment:
For absent only name field is required, why do you need to add all other fields in the example? The same for other modules such as credential, decision_environment, project, and user.

Reply:
Yeah the reason I had done it that way, I kept those other fields as required fields. Now I have removed it and you can delete all objects with just name and state as "absent"